### PR TITLE
letsencrypt: add IONOS DNS provider support

### PR DESCRIPTION
**Pull request by morphis**
IONOS provides API support to change DNS configuration which is supported by certbot. See https://developer.hosting.ionos.de/docs/dns and https://github.com/helgeerbe/certbot-dns-ionos

### CODE REVIEW

File: letsencrypt/build.yaml [hunk 1]
```diff
@@ -18,3 +18,4 @@ args:
   CERTBOT_NETCUP_VERSION: 1.1.1
   CERTBOT_NJALLA_VERSION: 1.0.0
   CERTBOT_VERSION: 1.21.0
+  CERTBOT_DNS_IONOS_VERSION: 2022.5.15
```
Comment (mdegat01) [inline letsencrypt/build.yaml:21]:
```suggestion
  CERTBOT_DNS_IONOS_VERSION: 2022.11.24
```
Is there a particular reason we aren't picking the current latest version here?

File: letsencrypt/DOCS.md [hunk 2]
```diff
@@ -89,6 +89,9 @@ gandi_api_key: ''
 gandi_sharing_id: ''
 transip_username: ''
 transip_api_key: ''
+ionos_endpoint: ''
```
Comment (mdegat01) [inline letsencrypt/DOCS.md:92]:
```suggestion
```
We should not make this an option. I know that the library makes it an option but I would expect users rarely ever change it. We should hard-code the endpoint in the addon to `https://api.hosting.ionos.com` and not make users look that information up. Unless you have reason to believe users will need to change this a lot, in which case please let me know why that is.

File: letsencrypt/rootfs/etc/services.d/lets-encrypt/run [hunk 3]
```diff
@@ -119,6 +119,13 @@ elif [ "${CHALLENGE}" == "dns" ] && [ "${DNS_PROVIDER}" == "dns-azure" ]; then
     fi
     PROVIDER_ARGUMENTS+=("--authenticator" "${DNS_PROVIDER}" "--${DNS_PROVIDER}-config" "/data/azure_creds")
 
+# IONOS
+elif [ "${CHALLENGE}" == "dns" ] && [ "${DNS_PROVIDER}" == "dns-ionos" ]; then
+    bashio::config.require 'dns.ionos_endpoint'
+    bashio::config.require 'dns.ionos_prefix'
+    bashio::config.require 'dns.ionos_secret'
+    PROVIDER_ARGUMENTS+=("--authenticator" "${DNS_PROVIDER}" "--${DNS_PROVIDER}-credentials" /data/dnsapikey)
```
Comment (mdegat01) [inline letsencrypt/rootfs/etc/services.d/lets-encrypt/run:127]:
```suggestion
    bashio::config.require 'dns.ionos_prefix'
    bashio::config.require 'dns.ionos_secret'
    PROVIDER_ARGUMENTS+=("--authenticator" "${DNS_PROVIDER}" "--${DNS_PROVIDER}-credentials" /data/dnsapikey "--${DNS_PROVIDER}-propagation-seconds" "${PROPAGATION_SECONDS}")
```
Need to support the propagation seconds option. Also removed the endpoint config option as mentioned above.

**Review by mdegat01 — CHANGES_REQUESTED** [review]
In addition to the listed changes you need to rebase.

You also need to bump the version in `config.yaml`. You added a new entry to the changelog but didn't change the version of the addon to match it.

**Comment by jandelgado** [comment]
@morphis , any chance we get this MR successfully merged? It seems to be already the 5th attempt, after #2331, #2626 (my shot), #2718 and #2721 - so finally getting this feature out of the door would be highly appreciated. Thanks for your efforts, hang on!

**Comment by gomble** [comment]
Hello when will this PR merged ?

**Comment by gomble** [comment]
@morphis can you have a look into your PR and correct the suggestions?

**Review by gomble — APPROVED** [review]

**Comment by sblu** [comment]
@morphis It would be great if you can get this PR in.

**Comment by mido1978** [comment]
Hi guys, 
is this topic closed and away now? You would me (and others) very happy, if you could bring IONOS into this.

Many thanks!

**Comment by opacey** [comment]
I'm very keen to have dns-ions added to the package, is it just a matter of finding someone with time to review and merge it?

**Comment by port3306** [comment]
Why is the status closed? Would be nice if someone could review and merge it.

**Comment by agners** [comment]
@port3306 the PR got closed because it got stale. The author did not address the review comments.

Unless the original author revives this PR, the common process for another author would be to pick the changes from here, address review feedback and open a new PR.

**Pull request closed without merging**